powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/database/repository/OperationRepository.java

```diff
@@ -47,8 +47,13 @@ public interface OperationRepository extends CrudRepository<OperationEntity, Str
     @Query("SELECT o FROM OperationEntity o WHERE o.id = :operationId")
     Optional<OperationEntity> findOperation(String operationId);
 
-    @Query("SELECT DISTINCT o FROM OperationEntity o INNER JOIN o.applications a WHERE o.userId = :userId AND a.id in :applicationIds ORDER BY o.timestampCreated DESC")
-    Stream<OperationEntity> findAllOperationsForUser(String userId, List<String> applicationIds, final Pageable pageable);
+    @Query("SELECT DISTINCT o FROM OperationEntity o INNER JOIN o.applications a INNER JOIN o.activation act " +
```

Comment:
ok I take this will take some time to crack. Maybe a split to several methods will be needed.